### ISSUE TITLE
can not call the deinit method

### ISSUE THREAD

**Issue by perfectChao** [SnapKit/SnapKit#209]
### New Issue Checklist

- [y ] I have looked at the [Documentation](http://snapkit.io/docs)
- [ ] I have read the [F.A.Q.](http://snapkit.io/faq)

##### Platform

* [ ] OS X
* [y ] iOS

##### Versions

* Platform version is: `INSERT PLATFORM VERSION`
* SnapKit version is: `INSERT SNAPKIT VERSION`

##### Integration

- [ y] Carthage
- [ ] Cocoapods
- [ ] Manually / Other

### Issue
    var tableView: UITableView!

    private func setupUI() {
        tableView = UITableView(frame: CGRectZero, style: .Grouped)
        tableView.delegate = self
        tableView.dataSource = self
        tableView.registerClass(SP_InputTextCell.self, forCellReuseIdentifier: SP_InputTextCellIndentfier)
        tableView.registerClass(SP_AddPicsCell.self, forCellReuseIdentifier: SP_SelectPicsCellIndentfier)
        self.view.addSubview(tableView)
        
        self.addMas()
    }
    
    private func addMas() {
        tableView.snp_makeConstraints { [unowned self](make) -> Void in
            make.edges.equalTo(self.view).inset(UIEdgeInsetsZero)
        }
    }`
 in the viewcontroller can not call the deinit method

`

**Comment by robertjpayne** [comment]
@perfectChao Sorry it's hard to follow what your asking, your code looks fine? I'm not entirely sure why your deinit method isn't being called but it's likely not SnapKit as none of SnapKit's closures are held onto past the scope they are used in.

**Comment by tc1993** [comment]
tableView.snp_makeConstraints { [weak self](make) -> Void in
        guard let self = self else {return}
        make.edges.equalTo(self.view).inset(UIEdgeInsetsZero)
    }

Changing to this would work, but I don't know why
